### PR TITLE
Catch and log any exception thrown in the checkin event method.

### PR DESCRIPTION
**Pull request by servingUpAces**
We don't want an exception while recording metrics to affect a user action. A KeyError exception was thrown today, that meant that a user say a 500, the action being performed was to download a document from the document-download-frontend app. By catching the error we prevent the user from seeing a 500 when a recording the connection metric fails.

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -387,14 +387,17 @@ def checkout(dbapi_connection, connection_record, connection_proxy):
...
+                # duration that connection was held by a single web request
+                duration = time.monotonic() - connection_record.info['checkout_at']
+
+                DB_CONNECTION_OPEN_DURATION_SECONDS.labels(
+                    connection_record.info['request_data']['method'],
+                    connection_record.info['request_data']['host'],
+                    connection_record.info['request_data']['url_rule']
+                ).observe(duration)
+            except Exception:
+                current_app.logger.exception("Exception caught for checkin event.")
```
Comment (idavidmcdonald) [inline app/__init__.py:403]: Can we log the original exception message so we have some context if we see this?

**Review by idavidmcdonald — APPROVED** [review]
Seems reasonable. Could argue that we should have a similar statement on the `checkout` but we can cross that bridge if we have to come to it

File: app/__init__.py (hunk 1 above)
Comment (servingUpAces) [reply app/__init__.py:403]: logger.exception will do that

Comment (idavidmcdonald) [reply app/__init__.py:403]: Ah OK, thanks

**Pull request merged**